dll/win32/shell32/shfldr.h

```diff
@@ -47,11 +47,50 @@ For Details See:
 #define SHFSF_COL_FATTS         4 // File attributes
 #define SHFSF_COL_COMMENT       5
 
+#define MAXREQUIREDREGITEMS 3
+typedef struct {
+    REFCLSID clsid;
+    LPCSTR pszCpl;
+    BYTE Order; // According to Geoff Chappell, required items have a fixed sort order
+} REQUIREDREGITEM;
```

Comment:
By the way, is this your own private shell structure?

Reply:
Yes, I don't know the real name of it nor what it contains other than the sort order which is not currently used (but will be later).

Reply:
From your answer it seems to imply it's a structure whose binary data layout is exposed somehow (either via exported function, or data stored in the registry...) ; is this the case?

Reply:
Neither? I think it is used like I'm doing here (passed directly to RegFolder and nowhere else). I assume these required items are "required" because if the registry gets corrupted, you still want My Computer and Control Panel to work.

Reply:
makes sense